google_api_lib/sync_tasks.py

```diff
@@ -18,12 +19,13 @@ def auth_allowed(backend, details, response, *args, **kwargs):
     Part of the SOCIAL_AUTH_PIPELINE (see settings.py).
     """
     email = details.get("email").lower()
-    # allow all emails if Google Drive integration is not set up
-    # otherwise, only allow emails added to Google Drive folder
-    if settings.GOOGLE_API_AUTHN_INFO and email not in get_file_user_emails(
-        settings.GOOGLE_DRIVE_HUNT_FOLDER_ID
-    ):
-        raise AuthForbidden(backend)
+    # Allow all emails if Google Drive integration is not set up
+    # or folder is world readable. Otherwise, only allow emails
+    # added to the Google Drive folder.
+    if settings.GOOGLE_API_AUTHN_INFO:
+        user_emails = get_file_user_emails(settings.GOOGLE_DRIVE_HUNT_FOLDER_ID)
+        if user_emails and email not in user_emails:
```

Comment:
Hmmhmm should we at least log a warning or error? The other thing is that this doesn't distinguish between the empty list and None, maybe we should check `if user_emails is not None` instead

Reply:
I added a log.warn() in my update, though when AuthForbidden is raised, that also already gets logged along with a stack trace.

I also updated to check `user_emails is not None`. In practice, I don't think `user_emails` will ever be an empty list because a folder always has an owner, so I think the list will always have at least one email. I update the docs for get_file_user_emails to explain the behavior (either a list of emails or None if the file is world-readable).

Reply:
>In practice, I don't think user_emails will ever be an empty list because a folder always has an owner, so I think the list will always have at least one email.

Oh yeah, you're right, lol. Thanks.

>I added a log.warn() in my update, though when AuthForbidden is raised, that also already gets logged along with a stack trace.

I meant log a warning if `user_emails` is None; it might be unexpected that the Cardboard instance is de facto public and that anyone with the URL can log in.

Reply:
Ooh, gotcha. I updated the log message to warn when the Google Drive folder is world-readable and thus anyone can log in to the Cardboard instance.